### PR TITLE
K medoids support

### PR DESCRIPTION
**Pull request by anjali-rgpt**
Pull request fixes ISSUE #92 
- Updated requirements document with the version for scikit-learn-extra
- Imported and added KMedoids algorithm into the model dictionary (attributes still need to be specified as per sklearn_extra docs)
   Here, for this clustering algorithm, tolerance and n_init must be removed from yaml and replaced with metric (which can be euclidean, cosine, manhattan, etc.) and method ('alternate' is the default method optimized for speed, 'pam' is more accurate in clustering but slower)
- Tested with existing dataset
- Added a conditional statement, where if the model class name is KMedoids, since KMedoids does not have a score function, the underlying inertia_ attribute is used (negative of inertia gives score, as per docs?)

---EDIT---

- Removed external dependency
- Implemented KMedoids class in extras folder based on original implementation, with minor changes for ease of documentation
- Only PAM algorithm has been implemented with two initialization methods: random, and heuristic
- Tested with yaml file on both fit and predict options
- Added required copyright from original source code

### CODE REVIEW

**Comment by Sahma61** [comment]
Hey @nidhaloff is the issue still open; if yes, could I take it?

**Comment by anjali-rgpt** [comment]
@nidhaloff I'm currently working on a K Medoids class, as per your suggestion. Parts of the code I'm using are licensed under 3-Clause BSD so I will just include the copyright in the docstring.

**Comment by anjali-rgpt** [comment]
@nidhaloff  I have made the necessary corrections. Do let me know if there is anything else to be done!

**Comment by nidhaloff** [comment]
@anjali-rgpt Looks good, thanks for your work ;)

**Comment by anjali-rgpt** [comment]
@nidhaloff  Thanks for all the advice! I might just implement another class for K-Medians (alternative to KMeans which uses Manhattan distance) if necessary,

**Comment by nidhaloff** [comment]
@anjali-rgpt Thanks! 
feel free to add any machine learning model. This is the point of the project, making machine learning easily accessible to everyone.
